src/futures/mod.rs

```diff
@@ -0,0 +1,9 @@
+//! Asynchronous APIs
+//!
+//! This traits use `core::future::Future` and generic associated types.
```

Comment:
The whole `futures` approach should be documented here. Including examples, executors, waking, etc.

Reply:
I agree with those — that being said I don't think it's quite sorted out how HALs are going to support this. I think they definitely can (and will), but it's going to require some thought as to how HALs should install interrupt handlers and whatnot for async operations to work.

Reply:
Sure. There is no "_one true answer_" as for how all operations should concert in all situations. However, the different alternatives and considerations about how to _implement_ and _consume_ these traits as well as what the expectations are about how the operations will _run_ should be explained here.